source/common/http/filter/ip_tagging_filter.cc

```diff
@@ -9,7 +13,45 @@ IpTaggingFilter::~IpTaggingFilter() {}
 
 void IpTaggingFilter::onDestroy() {}
 
-FilterHeadersStatus IpTaggingFilter::decodeHeaders(HeaderMap&, bool) {
+FilterHeadersStatus IpTaggingFilter::decodeHeaders(HeaderMap& headers, bool) {
+  bool is_internal_request =
+      headers.EnvoyInternalRequest() && (headers.EnvoyInternalRequest()->value() ==
+                                         Headers::get().EnvoyInternalRequestValues.True.c_str());
+
+  if ((is_internal_request && config_->requestType() == FilterRequestType::EXTERNAL) ||
+      (!is_internal_request && config_->requestType() == FilterRequestType::INTERNAL) ||
+      (!config_->runtime().snapshot().featureEnabled("ip_tagging.http_filter_enabled", 100))) {
+    return FilterHeadersStatus::Continue;
+  }
+
+  std::vector<std::string> tags =
+      config_->trie().getTags(callbacks_->requestInfo().downstreamRemoteAddress());
+
+  if (!tags.empty()) {
+    HeaderString& header = headers.insertEnvoyIpTags().value();
+    if (!header.empty()) {
+      header.append(", ", 2);
+    }
+
+    // When there are multiple tags, they are joined into one string prior to setting
+    // the header.
+    if (tags.size() > 1) {
+      const std::string tags_join = absl::StrJoin(tags, ", ");
+      header.append(tags_join.c_str(), tags_join.size());
+    } else {
+      header.append(tags[0].c_str(), tags[0].size());
+    }
```

Comment:
The append logic seems like a good thing to have a utility/utilities for.  I see at least two other places we do this comma logic (conn_manager_utility - forwarded client cert and xff appending)

I'd also lean towards no extra whitespace for the util but your call.